utils/db/setupRepos.ts

```diff
@@ -7,155 +8,52 @@ export interface SetupReposArgs {
-
-    const existingReposResult = await conn.query(query, [repo_owner, repo_provider, install_id]).catch(err => {
-        console.error(`[removeExtraRepositories] Could not get existing repositories from db for: ${repo_provider}/${repo_owner} for install_id: ${install_id}`, { pg_query: query }, err);
-        throw new Error('Failed to get existing repos from db');
-    });
-
-    const existingRepos = existingReposResult.rows.map(row => row.repo_name);
-    const extraRepos = existingRepos.filter(repo => !incomingRepoNames.includes(repo));
-
-    for (const repo_name of extraRepos) {
-        // Check if the current extra repo is the only one associated with the install_id
-        const singleRepoQuery = `SELECT install_id FROM repos WHERE repo_name = $1 AND repo_owner = $2 AND repo_provider = $3`;
-        const singleRepoResult = await conn.query(singleRepoQuery, [repo_name, repo_owner, repo_provider]).catch(err => {
-            console.error(`[removeExtraRepositories] Error checking if repo is the only one: ${repo_provider}/${repo_owner}/${repo_name} for install_id ${install_id}`, { pg_query: singleRepoQuery }, err);
-            throw new Error('Failed to check if repo is the only one associated with install_id');
-        });
-
-        const install_ids = singleRepoResult.rows[0]?.install_id || [];
-        const updatedInstallIds = install_ids.filter((id: String) => id !== install_id);
-        await conn.query('UPDATE repos SET install_id = $1 WHERE repo_name = $2 AND repo_owner = $3 AND repo_provider = $4', [updatedInstallIds, repo_name, repo_owner, repo_provider])
-        .catch(err => {
-            console.error(`[removeExtraRepositories] Could not remove install_id from extra repo: ${repo_provider}/${repo_owner}/${repo_name} for install_id ${install_id}`, { pg_query: query }, err);
-            throw new Error('Failed to remove install_id from extra repo');
-        });
+        await conn.query('COMMIT');
+        console.debug(`[saveSetupRepos] setup repos info saved successfully in db`)
+        return true;
+    } catch (err) {
+        await conn.query('ROLLBACK');
+        console.error(`[saveSetupRepos] Could not save setup repos for: ${install_id}, ${repo_names}`, { pg_query: insertReposQuery }, err);
+        return false;
     }
-
-    await conn.query('COMMIT').catch(err => {
-        console.error(`[removeExtraRepositories] Could not commit DB transaction: ${repo_provider}/${repo_owner} for install_id ${install_id}`, { pg_query: query }, err);
-        throw new Error('Failed to commit db transaction');
-    });
 };
 
-export const removePreviousInstallations = async (install_id: string) => {
+export const removePreviousInstallations = async (install_id: string, provider: string) => {
     const query = `UPDATE repos
     SET install_id = array_remove(install_id, '${install_id}')
-    WHERE '${install_id}' = ANY (install_id);
+    WHERE '${install_id}' = ANY (install_id) AND repo_provider = '${provider}';
```

Comment:
Low LOGAF.
Why do we need the repo_provider filter here? Is it because we were removing GitHub installations when a new Bitbucket installation was initiated?